compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -0,0 +1,1138 @@
+        return this.visit(node.parent());
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(IndexedExpressionNode node) {
+        BLangNode bLangNode = nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
+        if (bLangNode == null) {
+            return Optional.empty();
+        }
+
+        return getExpectedType(bLangNode);
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(RemoteMethodCallActionNode node) {
+        BLangNode bLangNode = nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
+        if (bLangNode == null) {
+            return Optional.empty();
+        }
+
+        if (bLangNode instanceof BLangInvocation.BLangActionInvocation) {
+            BLangInvocation.BLangActionInvocation bLangActionInvocation = (BLangInvocation.BLangActionInvocation)
+                    nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
+            if (isWithinParenthesis(linePosition, node.openParenToken(), node.closeParenToken())) {
+                if (bLangActionInvocation.argExprs.size() == 0) {
+                    return getTypeFromBType(((BInvokableSymbol) bLangActionInvocation.symbol).
+                            getParameters().get(0).getType());
+                }
+
+                Optional<BLangExpression> argument = bLangActionInvocation.argExprs.stream().filter
+                        (argumentNode -> isPosWithinRange(linePosition, argumentNode.getPosition()
+                                .lineRange())).findFirst();
+                if (argument.isPresent()) {
+                    return getExpectedType(argument.get());
+                }
+            }
+
+            return getExpectedType(bLangNode);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(ListConstructorExpressionNode node) {
+        BLangNode bLangNode = nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
+        if (bLangNode == null) {
+            return Optional.empty();
+        }
+
+        Optional<TypeSymbol> extractedType = getExpectedType(bLangNode);
+        if (!extractedType.isPresent()) {
+            return Optional.empty();
+        }
+
+        if (extractedType.get().typeKind() != TypeDescKind.ARRAY) {
+            throw new IllegalStateException();
+        }
+
+        if (isWithinParenthesis(linePosition, node.openBracket(), node.closeBracket())) {
+            BArrayType bArrayType = (BArrayType) ((BallerinaArrayTypeSymbol) extractedType.get()).getBType();
+            if (bArrayType.eType != null && bArrayType.eType.getKind() != TypeKind.OTHER) {
+                return Optional.of(typesFactory.getTypeDescriptor(bArrayType.eType));
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(CaptureBindingPatternNode node) {
+        Optional<Symbol> variableSymbol = this.getSymbolByName(node.variableName().text());
```

Comment:
It seems like we don't use the `variableSymbol` anywhere else except the `isEmpty` check. Shall we call `getSymbolByNode` within the if condition without initializing a value for it?